tests/components/rest/test_sensor.py

```diff
@@ -1,696 +1,665 @@
+                "name": "foo",
+                "unit_of_measurement": DATA_MEGABYTES,
+                "verify_ssl": "true",
+                "timeout": 30,
+            }
+        },
+    )
+    await hass.async_block_till_done()
+    assert len(hass.states.async_all()) == 1
+    state = hass.states.get("sensor.foo")
+
+    assert state.state == "bogus"
+    assert state.attributes["led0"] == "0"
+    assert state.attributes["led1"] == "0"
+    assert state.attributes["temp0"] == "0x0XF0x0XF"
+    assert state.attributes["time0"] == "0"
+    assert state.attributes["ver"] == "12556"
+
+
+@respx.mock
+async def test_update_with_application_xml_convert_json_attrs_with_jsonattr_template(
+    hass,
+):
+    """Test attributes get extracted from a JSON result that was converted from XML with application/xml mime type."""
+
+    respx.get(
+        "http://localhost",
+        status_code=200,
+        headers={"content-type": "application/xml"},
+        content="<main><dog>1</dog><cat>3</cat></main>",
+    )
+    assert await async_setup_component(
+        hass,
+        "sensor",
+        {
+            "sensor": {
+                "platform": "rest",
+                "resource": "http://localhost",
+                "method": "GET",
+                "value_template": "{{ value_json.main.dog }}",
+                "json_attributes_path": "$.main",
+                "json_attributes": ["dog", "cat"],
+                "name": "foo",
+                "unit_of_measurement": DATA_MEGABYTES,
+                "verify_ssl": "true",
+                "timeout": 30,
+            }
+        },
+    )
+    await hass.async_block_till_done()
+    assert len(hass.states.async_all()) == 1
+    state = hass.states.get("sensor.foo")
+
+    assert state.state == "1"
+    assert state.attributes["dog"] == "1"
+    assert state.attributes["cat"] == "3"
+
+
+@respx.mock
+async def test_update_with_xml_convert_bad_xml(hass, caplog):
+    """Test attributes get extracted from a XML result with bad xml."""
+
+    respx.get(
+        "http://localhost",
+        status_code=200,
+        headers={"content-type": "text/xml"},
+        content="",
+    )
+    assert await async_setup_component(
+        hass,
+        "sensor",
+        {
+            "sensor": {
+                "platform": "rest",
+                "resource": "http://localhost",
+                "method": "GET",
+                "value_template": "{{ value_json.toplevel.master_value }}",
+                "json_attributes": ["key"],
+                "name": "foo",
+                "unit_of_measurement": DATA_MEGABYTES,
+                "verify_ssl": "true",
+                "timeout": 30,
+            }
+        },
+    )
+    await hass.async_block_till_done()
+    assert len(hass.states.async_all()) == 1
+    state = hass.states.get("sensor.foo")
+
+    assert state.state == STATE_UNKNOWN
+    assert "Erroneous XML" in caplog.text
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)